src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_help.py

```diff
@@ -72,17 +72,83 @@
     type: group
     short-summary: Generic commands to manage Azure resources
 """
+helps['resource list'] = """
+    type: command
+    short-summary: list resource
+    examples:
+        - name: list all resource in a region
+          text: >
+            az resource list --location westus
+        - name: list resource with a name
+          text: >
+            az resource list --name thename
+        - name: list resources with a tag
+          text: >
+             az resource list --tag something
+        - name: list resource with a tag with a particular prefix
+          text: >
+            az resource list --tag some*
+        - name: list resource with a tag value
+          text: >
+            az resource list --tag something=else
+"""
+
 helps['resource show'] = """
     type: command
     short-summary: display a resource detail
+    long-summary: Get details of a resource like /subscriptions/0000/resourceGroups/myGroup/providers/Microsoft.Provider/resA/myA/resB/myB/resC/myC
     examples:
         - name: show a virtual machine
           text: >
             az vm show -g mygroup -n myvm --resource-type "Microsoft.Compute/virtualMachines"
-        - name: show a webapp
+        - name: show a webapp using resource id
+          text: >
+            az resource show --id /subscriptions/0b1f6471-1bf0-4dda-aec3-111111111111/resourceGroups/myGroup/providers/Microsoft.Web/sites/myWebapp
+        - name: show a subnet
+          text: >
+            az resource show -g mygroup -n mysubnet --namespace microsoft.network --parent virtualnetworks/myvnet --resource-type subnets
+        - name: show a subnet using id
+          text: >
+            az resource show --id /subscriptions/0b1f6471-1bf0-4dda-aec3-111111111111/resourceGroups/myGroup/providers/Microsoft.Network/virtualNetworks/myvnet/subnets/mysubnet
+        - name: show an app gateway path rule
           text: >
-            az resource show -g mygroup -n mywebapp --resource-type "Microsoft.web/sites"
+            az resource show -g myGroup --namespace Microsoft.Network --parent applicationGateways/ag1/urlPathMaps/map1 --resource-type pathRules -n rule1  
```

Comment:
We want to move away from this "parent" parameter - only makes it harder for users to identify a resource.